docs: Remove onramp kit from onramp guides (#553)
safe-global/safe-docs: pages/sdk/onramp/monerium.mdx

import { Steps } from 'nextra/components'
import Image from 'next/image'
import MoneriumAddMoney from '../../../assets/monerium-sandbox-add-money.png'

# Onramping with Monerium

The [Monerium SDK](https://www.npmjs.com/package/@monerium/sdk) enables Safe users to directly transfer e-money tokens from their Safe accounts to an IBAN via the SEPA network.

> More info about Monerium:

Author: Yes! Should I remove the link in the paragraph, or is it good to have a duplicated link?

> More info about Monerium:

Reviewer: Best to not have duplicates to avoid content debt.

> More info about Monerium:

Reviewer: This was not resolved, we should link the mentions in the paragraph above instead of dangling links.

> More info about Monerium:

Author: Ok, now I am a little confused. What do you prefer? To add it to a 'further reading' section, or to leave it in the paragraph?

  chainId: 11155111 // Chain ID of Sepolia in this example
})
```

Calling `authorize` will redirect the user to the Monerium login page. 

### Authenticate with Monerium

At Monerium, the user needs to log in or create an account. 
Once logged in, Monerium will verify the ownership of the Safe by checking the signed message. 
After successful verification, Monerium will create an IBAN and link it to the Safe. 
The user will then be redirected to the specified `redirectUrl` with an added `code` query parameter.

### Finish the authentication

Once the user lands back on your page, finish the authentication process.

```typescript
// Returns true, if the user is authorized
const isAuthorized = await monerium.getAccess()
```

Congratulations, you authenticated your user with Monerium and linked the Safe to the Monerium account.
</Steps>

## Step 2: Place an Order

Now that the user is authenticated with Monerium, you can place an order to transfer tokens from the Safe to an IBAN.

<Steps>

### Get some tokens

First, your user needs to obtain some EURe test tokens on Sepolia:

* Log into the [Monerium Sandbox](https://sandbox.monerium.dev/accounts) account.

> * Click on the `Add Money` button.

Reviewer: The button in the attached screenshot says "ADD MONEY".

// Initialize the Safe\{Core\} SDK and link it to an existing Safe
const protocolKit = await Safe.init({
  provider: RPC_URL, // set a valid RPC URL
  signer: OWNER_1_PRIVATE_KEY, // set the private key of the first Safe owner
  safeAddress
})

// Create a signed message by creating a transaction to the signMessage contract
const signMessageContract = await getSignMessageLibContract({
  safeProvider: protocolKit.getSafeProvider(),
  safeVersion: await protocolKit.getContractVersion()
})

// Let the contract encode the message's hash to get the transaction data
const txData = signMessageContract.encode('signMessage', [
  hashSafeMessage(constants.LINK_MESSAGE) // 'I hereby declare that I am the address owner.'
])

// Assemble a transaction object
const safeTransactionData = {
  to: await signMessageContract.getAddress(),
  value: '0',
  data: txData,
  operation: OperationType.DelegateCall
}

// Create a transaction
const signMessageTx = await protocolKit.createTransaction({
  transactions: [safeTransactionData]
})

// Sign the transaction with the first owner
const signedTx = await protocolKit.signTransaction(signMessageTx)

// Connect the protocol kit to the second owner

> const protocolKitOfOwner2 = await protocolKit.connect({

Author: That is a very good point! I am on the brink. On one hand, you are right, and the confirmation happens on the second owner's device. On the other hand, this makes the code more complex, and the focus of this tutorial is the onramping with Monerium. I am like 51% leaning towards leaving this as it is, but I am open to changing it.

> const protocolKitOfOwner2 = await protocolKit.connect({

Reviewer 2: OKay, let's leave it as it is.

After the message is signed and the transaction is executed, the user can authenticate with Monerium.

```typescript
import { MoneriumClient } from '@monerium/sdk'

// Initialize the Monerium Client
const monerium = new MoneriumClient({
  clientId: 'a1b2c3-x7y8y9', // Get your client ID from Monerium
  environment: 'sandbox' // Use the appropriate Monerium environment ('sandbox' | 'production')
})

// Start the Monerium authentication flow and send the user to Monerium
await monerium.authorize({
  address: safeAddress, // The address of the user's Safe
  signature: '0x', // '0x' for Safe authentication lets Monerium look for the signature on-chain
  redirectUrl: 'http://localhost:3000/return', // URL where Monerium will redirect the user after authenticating
  chainId: 11155111 // Chain ID of Sepolia in this example
})
```

Calling `authorize` will redirect the user to the Monerium login page. 

### Authenticate with Monerium

At Monerium, the user needs to log in or create an account. 
Once logged in, Monerium will verify the ownership of the Safe by checking the signed message. 
After successful verification, Monerium will create an IBAN and link it to the Safe. 

> The user will then be redirected to the specified `redirectUrl` with an added `code` query parameter.

Author: I replaced the text with "The users will then be redirected to the specified `redirectUrl` with the new session id as a GET parameter." wdyt?

> The user will then be redirected to the specified `redirectUrl` with an added `code` query parameter.

Reviewer 2: Sounds good!

### Finish the authentication

Once the user lands back on your page, finish the authentication process.

```typescript
// Returns true, if the user is authorized
const isAuthorized = await monerium.getAccess()
```

Congratulations, you authenticated your user with Monerium and linked the Safe to the Monerium account.
</Steps>

## Step 2: Place an Order

Now that the user is authenticated with Monerium, you can place an order to transfer tokens from the Safe to an IBAN.

<Steps>

### Get some tokens

First, your user needs to obtain some EURe test tokens on Sepolia:

* Log into the [Monerium Sandbox](https://sandbox.monerium.dev/accounts) account.
* Click on the `ADD MONEY` button.
* Create a test IBAN transfer onto the account. The tokens from this test transfer will be available in the Safe.

<Image src={MoneriumAddMoney} width={300} height={500}  />

### Send an order to Monerium

> To send tokens from the Safe to an IBAN, the user must sign a message with the Safe and send the order to Monerium. 
> You can trigger both actions simultaneously.

Reviewer 2: I would rephrase it this way for simplicity and coherence with the order of our steps:
```suggestion
To send tokens from a Safe to an IBAN, users must send the order to Monerium and sign a message with their account.
```

```typescript
import { placeOrderMessage } from '@monerium/sdk'

const amount = '10' // Specify the amount in Euro
const iban = 'AL47212110090000000235698741' // The target IBAN

// 'Send EUR 10 to AL47212110090000000235698741 at Fri, 17 May 2024 20:55:29Z'
const orderMessage = placeOrderMessage(amount, 'eur', iban)

// Send the order to the Monerium backend
const order = await moneriumClient.placeOrder({
  amount,
  signature: '0x',
  currency: 'eur',
  address: safeAddress, // the Safe address
  counterpart: {
    identifier: {
      standard: 'iban',
      iban
    },
    details: {
      firstName: 'User',
      lastName: 'Userson',

>       county: 'AL'

Author: Thanks, it actually is County. 

<img width="834" alt="grafik" src="https://github.com/user-attachments/assets/0586c4a3-8f99-4c7e-8b9e-67acfe3680e2"> https://www.npmjs.com/package/@monerium/sdk

    }
  },
  message: orderMessage,
  memo: 'Powered by Monerium SDK',
  chain: 'ethereum',
  network: 'sepolia'
})
```

> Monerium will listen for the sign message transaction of the following step on the chain and execute the order once the transaction settles.

Reviewer 2: ```suggestion
Monerium will listen for the sign message transaction of the following step on the selected chain and execute the order once the transaction settles.
```
